File: src/main/java/server/mecs/proxyservermanager/listeners/LoginListener.java
package server.mecs.proxyservermanager.listeners;

import net.md_5.bungee.api.ProxyServer;
import net.md_5.bungee.api.chat.ComponentBuilder;
import net.md_5.bungee.api.connection.ProxiedPlayer;
import net.md_5.bungee.api.event.PostLoginEvent;
import net.md_5.bungee.api.plugin.Listener;
import net.md_5.bungee.event.EventHandler;
import server.mecs.proxyservermanager.ProxyServerManager;
import server.mecs.proxyservermanager.threads.*;

import java.util.concurrent.ExecutionException;

public class LoginListener implements Listener {

    ProxyServerManager plugin = null;

    public LoginListener(ProxyServerManager plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onLogin(PostLoginEvent e) throws InterruptedException, ExecutionException {
        ProxiedPlayer player = e.getPlayer();

        PlayerData.PlayerData(plugin, player);

        if (CheckBanned.isBanned(plugin, player.getName())) {
            String reason = getBanReason.getBanReason(plugin, player.getName());
            player.disconnect(new ComponentBuilder("§cYou are permanently banned from this server.\n \n§7Reason: §f" + reason).create());
            return;
        }

        if (CheckMuted.isMuted(plugin, player.getName())) {
            plugin.MuteMap.put(player.getUniqueId(), true);
        }

        LoginLog.LoginLog(plugin, e.getPlayer());

        if (player.getUniqueId().toString().length() != 32){
            ProxyServer.getInstance().getPluginManager().dispatchCommand(ProxyServer.getInstance().getConsole(), "Your account has a security alert.");
        }

        if (player.hasPermission("group.owner") || player.hasPermission("group.administrator")) {
            ProxyServer.getInstance().broadcast(new ComponentBuilder("§b>§d>§b> §c" + e.getPlayer().getName() + "§6 joined the network.").create());
            return;
        }

        if (player.hasPermission("group.moderator")) {
            ProxyServer.getInstance().broadcast(new ComponentBuilder("§b>§d>§b> §2" + e.getPlayer().getName() + "§6 joined the network.").create());
            return;
        }

        if (player.hasPermission("group.helper")) {
            ProxyServer.getInstance().broadcast(new ComponentBuilder("§b>§d>§b> §9" + e.getPlayer().getName() + "§6 joined the network.").create());
            return;
        }

        if (player.hasPermission("group.youtuber")) {
            ProxyServer.getInstance().broadcast(new ComponentBuilder("§b>§d>§b> §f" + e.getPlayer().getName() + "§6 joined the network.").create());
            return;
        }

        if (player.hasPermission("group.vip+++") || player.hasPermission("group.vip++") || player.hasPermission("group.vip+") || player.hasPermission("group.vip")) {
            ProxyServer.getInstance().broadcast(new ComponentBuilder("§b>§d>§b> §a" + e.getPlayer().getName() + "§6 joined the network.").create());
            return;
        }

        ProxyServer.getInstance().broadcast(new ComponentBuilder("§b>§d>§b> §7" + e.getPlayer().getName() + "§6 joined the network.").create());

    }
}

File: src/main/java/server/mecs/proxyservermanager/listeners/LoginListener.kt
package server.mecs.proxyservermanager.listeners

import net.md_5.bungee.api.ProxyServer
import net.md_5.bungee.api.chat.ComponentBuilder
import net.md_5.bungee.api.event.PostLoginEvent
import net.md_5.bungee.api.plugin.Listener
import net.md_5.bungee.event.EventHandler
import server.mecs.proxyservermanager.ProxyServerManager
import server.mecs.proxyservermanager.ProxyServerManager.Companion.es
import server.mecs.proxyservermanager.database.MongoDBManager
import server.mecs.proxyservermanager.database.PlayerData.checkPlayerData
import server.mecs.proxyservermanager.getTime.getDate

class LoginListener(private val plugin: ProxyServerManager) : Listener {
    @EventHandler
    fun onLogin(e: PostLoginEvent) {
        val player = e.player
        es.execute {
            MongoDBManager(plugin, "PlayerData").use { con ->
                checkPlayerData(con, player)
            }
        }

        MongoDBManager.executeLoginQueue(
                "{'mcid':'${e.player.name}', " +
                "'uuid':'${e.player.uniqueId}', " +
                "'address':'${player.socketAddress}'" +
                "'date':'${getDate()}'}"
        )

        val prefix = when {
            player.hasPermission("group.owner") -> "§c[OWNER] "
            player.hasPermission("group.administrator") -> "§c[ADMIN] "
            player.hasPermission("group.moderator") -> "§2[MOD] "
            player.hasPermission("group.helper") -> "§9[HELPER] "
            player.hasPermission("group.youtuber") -> "§c[§f§lYT§c] "
            player.hasPermission("group.vip+++") -> "§a[VIP§0+++§a] "
            player.hasPermission("group.vip++") -> "§a[VIP§b++§a] "
            player.hasPermission("group.vip+") -> "§a[VIP§6+§a] "
            player.hasPermission("group.vip") -> "§a[VIP§a] "
            else -> "§7"
        }
        ProxyServer.getInstance().broadcast(*ComponentBuilder("§b>§d>§b> $prefix${player.name}§6 joined the network.").create())
    }
}